client/client_mock_test.go

```diff
@@ -21,18 +42,9 @@ func transportEnsureBody(f transportFunc) transportFunc {
 
 // WithMockClient is a test helper that allows you to inject a mock client for testing.
 func WithMockClient(doer func(*http.Request) (*http.Response, error)) Opt {
-	return func(c *clientConfig) error {
-		c.client = &http.Client{
-			Transport: transportEnsureBody(transportFunc(doer)),
-		}
-		if !c.manualOverride {
-			c.version = ""
-		}
-		c.host = ""
-		c.proto = ""
-		c.addr = ""
-		return nil
-	}
+	return WithHTTPClient(&http.Client{
+		Transport: transportEnsureBody(transportFunc(doer)),
+	})
```

Comment:
FWIW; on this one I was slightly going back-and-forth if I should open-code it, or use this as a wrapper for `WithHTTPClient` - it's a test-utility so we can still make up our mind if we prefer the other approach.